### PR TITLE
Fix Role Param Mismatch Issue

### PR DESCRIPTION
**Pull request by RayLiu7**
# Description
Fix Role Param Mismatch Issue 

# ISSUE TYPE
- Bugfix Pull Request

##### RESOURCE OR DATASOURCE NAME
Role Resource

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [x] Acceptance tests

### CODE REVIEW

**Comment by RayLiu7** [comment]
run e2e test

File: docs/resources/role.md [hunk 1]
```diff
@@ -60,18 +60,16 @@ resource "powerscale_role" "role_test" {
 
   # Optional fields both for creating and updating
   description = "role_test_description"
-  # To add members, please provide uid/gid or provide name and type
+  # To add members, the uid/gid is required. Please use user/user_group datasource to look up the uid/gid needed.
   members = [
     {
-      name = "admin",
-      type = "user"
+      id = "UID:10"
```
Comment (taohe1012) [inline docs/resources/role.md:66]: the id/name is convenient to user, we could keep them. And we can check name and id are matched when assign values

**Review by baoy1 — APPROVED** [review]

File: docs/resources/role.md (hunk 1 above)
Comment (RayLiu7) [reply docs/resources/role.md:66]: Hi @taohe1012 ,as discussed in scrum, we shall keep the id as required now. Thanks

**Comment by RayLiu7** [comment]
run e2e test

File: docs/resources/role.md (hunk 1 above)
Comment (taohe1012) [reply docs/resources/role.md:66]: this is not wrong error solution, but id/name is indeed more convenient to user.... approved

**Review by taohe1012 — APPROVED** [review]

**Pull request merged**